### ISSUE TITLE
Exchange arrows don't move when resizing the map horizontally

### ISSUE THREAD

**Issue by q--** [tmrowco/electricitymap-contrib#2573]
Resize the browser window horizontally. The map behaves correctly, but the arrows don't move. (They don't jump to the right spot after you're done resizing either.)

Resizing vertically works fine.

Occurs on both Chrome and Firefox, so it probably happens for all browsers.

**Comment by Kongkille** [comment]
Hi :) I just tried myself and I can't seem to replicate this. I see that it seems they do not move to the right spot while resizing the window, however when I am done resizing it seems to work. Can you perhaps attach a .gif or a video?

I've tested on Chrome & Safari.

**Comment by q--** [comment]
![ElectricityMap horizontal resize bug](https://user-images.githubusercontent.com/2470175/87421625-88ae8280-c5d7-11ea-96ab-d0095ea01019.gif)

**Comment by Kongkille** [comment]
Ah I see. I resized it using the bottom right corner and only dragged horizontally, my bad. Thanks for finding this :)

**Comment by fbarl** [comment]
Good catch @q--!

It seems that the source of this issue is that [react-map-gl](https://github.com/visgl/react-map-gl) `onViewportChange` hook gets called when the map viewport height gets updated, but _not when the width does_. I'm tempted to report an issue on their repo as I didn't find anything similar online, but before I do I'd like to make sure it's not something caused by our CSS :thinking: 

In any case, their is a simple workaround if we want to move fast - we can simply insert our own separate width listener in [layout/map.js](https://github.com/tmrowco/electricitymap-contrib/blob/aa523a67aa821b2005c7c94dc3cf55451c44c1c4/web/src/layout/map.js) - I tried this piece of code and it fixed the issue:

```javascript
const width = useWidthObserver(ref);

useEffect(() => {
  handleViewportChange({ ...viewport, width });
}, [width]);
```

**Comment by fbarl** [comment]
I opened visgl/react-map-gl#1157 to try to address the issue in the library, but in the meantime I addressed it via the above suggested hack in #2589.
